Providers/Profile.swift

```diff
@@ -703,15 +703,22 @@ public class BrowserProfile: Profile {
 
         // Used as a task queue for syncing.
         private var syncReducer: AsyncReducer<EngineResults, EngineTasks>?
+        private var syncStatsReport: SyncStatsReport?
```

Comment:
I'm not a huge fan of this shared mutable conditional state. Can we not thread this into a sync and back out the end as part of a result?